### PR TITLE
feat(kmodalfullscreen): vue 3 upgrade

### PR DESCRIPTION
**Pull request by kaiarrowood**
### Summary
Port `KModalFullscreen` component.

### Vue 3 Upgrade

**If your PR contains code that needs to be ported to the `beta` branch, please add the [`port to beta branch`](https://github.com/Kong/kongponents/labels/port%20to%20beta%20branch) label to your PR.**

If you have questions, tag `@adamdehaven` or `@kaiarrowood`.

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: docs/.vuepress/config.ts [hunk 1]
```diff
@@ -108,6 +108,7 @@ export default defineUserConfig<DefaultThemeOptions, ViteBundlerOptions>({
               '/components/label',
               '/components/menu',
               '/components/modal',
+              '/components/kmodalfullscreen',
```
Comment (adamdehaven) [inline docs/.vuepress/config.ts:111]:
```suggestion
              '/components/modal-fullscreen',
```
issue: wrong file name

File: docs/components/modal-fullscreen.md [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+  @canceled="defaultIsOpen = false"
+  @proceed="defaultIsOpen = false">
+  <p>{{ alotOfText }}</p>
+</KModalFullscreen>
+
+<script>
+export default {
+  data () {
+    return {
+      isVisible: false
```
Comment (adamdehaven) [inline docs/components/modal-fullscreen.md:35]: question: I don't think this is being used in the example..?

File: docs/.vuepress/config.ts (hunk 1 above)
Comment (kaiarrowood) [reply docs/.vuepress/config.ts]: Fixed

File: docs/components/modal-fullscreen.md (hunk 2 above)
Comment (kaiarrowood) [reply docs/components/modal-fullscreen.md]: Updated

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**